File: mw/app/controllers/ng/v1/vehicles_controller.rb
# Vehicles Controller that inherits from Base
class Ng::V1::VehiclesController < Ng::V1::BaseController
  before_action :set_vehicle, only: [:next_state]
  def index
    render json: Vehicle.all, status: :ok
  end

  def next_state
    if @vehicle.next_state!
      render json: { vehicle: @vehicle }, status: :ok
    else
      render json: { message: 'Failed to move to next state' }, status: :forbidden
    end
  end

  private

  def set_vehicle
    @vehicle = Vehicle.find(params[:id])
    render status: :not_found and return unless @vehicle
  end
end

File: ng_app/src/app/vehicles/vehicles.controller.js
(function() {
  'use strict';

  angular
    .module('ngApp')
    .controller('VehiclesController', VehiclesController);

  /** @ngInject */
  function VehiclesController($scope, Restangular, $log, $window, $state) {
    var vm = this

    vm.index = function(){
      Restangular.all('vehicles').getList().then(function(result) {
          vm.list = result
      },function(error) {
          $log.info(error.data.error_message)
          alert(error.data.error_message)
      });
    }

    vm.completeState = function(vehicle){
      var confirmRequest = $window.confirm('Are you sure you want to move the vehicle to the next state ');
      if(confirmRequest){
        Restangular.all('vehicles/' + vehicle.id + "/next_state").customPUT().then(function(result) {
          vehicle.state_name = result.vehicle.state_name
        },function(error) {
            $log.info(error)
            alert(error.data.message)
        });
      }
    }

    vm.goToAdmin = function(){
      $state.go("states")
    }
  }
})();
